yt_dlp/utils/_utils.py

```diff
@@ -167,6 +167,19 @@ def IDENTITY(x):
 
 NUMBER_RE = r'\d+(?:\.\d+)?'
 
+WINDOWS_RESERVED_NAMES = (
+    'CON', 'CONOUT$', 'CONIN$', 'PRN', 'AUX', 'NUL',
+    *tuple(f'{name:s}{num:d}' for name, num in itertools.product(('COM', 'LPT'), range(10))),
+    *tuple(
+        f'{name:s}{ssd:s}'
+        for name, ssd in itertools.product(
+            ('COM', 'LPT'),
+            ('\N{SUPERSCRIPT ONE}', '\N{SUPERSCRIPT TWO}', '\N{SUPERSCRIPT THREE}'),
+        )
+    ),
+)
+WINDOWS_RESERVED_NAMES_RE = fr'({"|".join(WINDOWS_RESERVED_NAMES)})'
```

Comment:
If `WINDOWS_RESERVED_NAMES` is not being used elsewhere, let's just make the RE in one go, also considering that `$` is a significant character in a RE:
```suggestion
WINDOWS_RESERVED_NAMES_RE = '|'.join((
    'CON', r'CON(?:IN|OUT)\$', 'PRN', 'AUX', 'NUL',
    r'(?:COM|LPT)[\d¹²³],
))
```
Or scrap the `join()` altogether, but I think it's clearer as above.
```

Reply:
I prefer the `\N{}` syntax for... personal reasons... I think the current code also needs a bit of a refactor, so that it just uses a single replace. Im also not a fan of appending `_res` as we are using a `#` instead in every other case

Reply:
As to `\N{}`, I can see arguments either way, but it makes replacement line 172 54 characters longer.